pkg/callback/scheduler_callback.go

```diff
@@ -118,15 +118,36 @@ func (callback *AsyncRMCallback) RecvUpdateResponse(response *si.UpdateResponse)
 	for _, release := range response.ReleasedAllocations {
 		log.Logger().Debug("callback: response to released allocations",
 			zap.String("UUID", release.UUID))
+
+		// delete the allocated pod
+		if app := callback.context.GetApplicationInternal(release.ApplicationID); app != nil {
+			// TerminationType 0 mean STOPPED_BY_RM
+			if release.TerminationType != si.AllocationRelease_STOPPED_BY_RM {
+				for _, task := range app.GetTaskMap() {
+					if task.GetTaskAllocationUUID() == release.UUID {
+						err := task.DeleteTaskPod(task.GetTaskPod())
+						if err != nil {
+							log.Logger().Error("failed to delete pod", zap.Error(err))
+						}
+					}
+				}
+			}
+		}
```

Comment:
let's use event handling to handle this, so we do not need to expose `GetApplicationInternal ()`,  `GetTaskMap()` functions. We can change this to:

```
ev := cache.NewReleaseAppAllocationEvent(app.GetApplicationID(), toReleaseAllocations, events.ReleaseAppAllocation)
dispatcher.Dispatch(ev)
```

in `application.go`, we just need to add another event `ReleaseAppAllocation`, and that triggers state transition from `Running` to `Running`. And in the handler, we can do the task pod cleanup.

Reply:
OK, done.